src/pages/workspaces/workspace/LockWorkspaceModal.js

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { ButtonPrimary, spinnerOverlay } from 'src/components/common'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import { reportError } from 'src/libs/error'
+
+
+const LockWorkspaceModal = ({ workspace: { workspace: { namespace, name, isLocked } }, onDismiss, onSuccess }) => {
+  const [togglingLock, setTogglingLock] = useState(false)
+  const helpText = isLocked ? 'Unlock Workspace' : 'Lock Workspace'
+
+  const toggleWorkspaceLock = async () => {
+    try {
+      setTogglingLock(true)
+      isLocked ? await Ajax().Workspaces.workspace(namespace, name).unlock() : await Ajax().Workspaces.workspace(namespace, name).lock()
+      onDismiss()
+      onSuccess()
+    } catch (error) {
+      reportError('Error toggling workspace lock', error)
+      setTogglingLock(false)
+    }
+  }
```

Comment:
We have some error handling wrappers that will handle the error reporting for you.
Also, this corrects an issue where `onDismiss` will not get called if the error is thrown resulting in the modal persisting if there is a network error.

```suggestion
  const onFailureDismiss = () => {
    setTogglingLock(false)
    onDismiss()
  }

  const toggleWorkspaceLock = withErrorReportingInModal('Error toggling workspace lock', onFailureDismiss, async () => {
    setTogglingLock(true)
    isLocked ? await Ajax().Workspaces.workspace(namespace, name).unlock() : await Ajax().Workspaces.workspace(namespace, name).lock()
    onDismiss()
    onSuccess()
  })
```